SignalMessaging/attachments/AttachmentApprovalViewController.swift

```diff
@@ -0,0 +1,171 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import MediaPlayer
+
+@objc
+public protocol AttachmentApprovalViewControllerDelegate: class {
+    func didApproveAttachment()
+    func didCancelAttachment()
+}
+
+@objc
+public class AttachmentApprovalViewController: OWSViewController {
+
+    let TAG = "[AttachmentApprovalViewController]"
+    weak var delegate: AttachmentApprovalViewControllerDelegate?
+
+    // MARK: Properties
+
+    let attachment: SignalAttachment
+
+    let mediaMessageView: MediaMessageView
+
+    // MARK: Initializers
+
+    @available(*, unavailable, message:"use attachment: constructor instead.")
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("unimplemented")
+    }
+
+    required public init(attachment: SignalAttachment, delegate: AttachmentApprovalViewControllerDelegate) {
+        assert(!attachment.hasError)
+        self.attachment = attachment
+        self.delegate = delegate
+        self.mediaMessageView = MediaMessageView(attachment: attachment, mode: .large)
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    // MARK: View Lifecycle
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.backgroundColor = UIColor.white
+
+        createViews()
+
+        self.navigationItem.title = dialogTitle()
+    }
+
+    private func dialogTitle() -> String {
+        guard let filename = mediaMessageView.formattedFileName() else {
+            return NSLocalizedString("ATTACHMENT_APPROVAL_DIALOG_TITLE",
+                                     comment: "Title for the 'attachment approval' dialog.")
+        }
+        return filename
+    }
+
+    override public func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        mediaMessageView.viewWillAppear(animated)
+    }
+
+    override public func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+
+        mediaMessageView.viewWillDisappear(animated)
+    }
+
+    // MARK: - Create Views
+
+    private func createViews() {
+        let previewTopMargin: CGFloat = 30
+        let previewHMargin: CGFloat = 20
+
+        self.view.addSubview(mediaMessageView)
+        mediaMessageView.autoPinWidthToSuperview(withMargin:previewHMargin)
+        mediaMessageView.autoPin(toTopLayoutGuideOf: self, withInset:previewTopMargin)
```

Comment:
* WhatsApp uses a "white background" contact picker view and a "black background" for the approval/caption/etc. view.  I'm inclined to use a white background for both.  
* WhatsApp's attachment approval view has a pretty small "send" button.  I wonder what our caption/sender UI will look like.  I'm not sure it should like like conversation view's, for example.  e.g. Conversation view's input toolbar "grows" as the text message grows in length.  I think this growth might be much more jarring in the context of attachment approval.

Thoughts?